src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -66,27 +66,76 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
         {
             if (Options.RemoteSignOutPath.HasValue && Options.RemoteSignOutPath == Request.Path)
             {
-                var remoteSignOutContext = new RemoteSignOutContext(Context, Options);
-                await Options.Events.RemoteSignOut(remoteSignOutContext);
+                return await HandleRemoteSignOutAsync();
+            }
+            return await base.HandleRequestAsync();
+        }
+
+        protected virtual async Task<bool> HandleRemoteSignOutAsync()
+        {
+            OpenIdConnectMessage message = null;
+
+            if (string.Equals(Request.Method, "GET", StringComparison.OrdinalIgnoreCase))
+            {
+                message = new OpenIdConnectMessage(Request.Query.Select(pair => new KeyValuePair<string, string[]>(pair.Key, pair.Value)));
+            }
+            // assumption: if the ContentType is "application/x-www-form-urlencoded" it should be safe to read as it is small.
+            else if (string.Equals(Request.Method, "POST", StringComparison.OrdinalIgnoreCase)
+              && !string.IsNullOrEmpty(Request.ContentType)
+              // May have media/type; charset=utf-8, allow partial match.
+              && Request.ContentType.StartsWith("application/x-www-form-urlencoded", StringComparison.OrdinalIgnoreCase)
+              && Request.Body.CanRead) {
+                var form = await Request.ReadFormAsync();
+                message = new OpenIdConnectMessage(form.Select(pair => new KeyValuePair<string, string[]>(pair.Key, pair.Value)));
+            }
+
+            var remoteSignOutContext = new RemoteSignOutContext(Context, Options, message);
+            await Options.Events.RemoteSignOut(remoteSignOutContext);
 
-                if (remoteSignOutContext.HandledResponse)
+            if (remoteSignOutContext.HandledResponse)
+            {
+                Logger.RemoteSignOutHandledResponse();
+                return true;
+            }
+            if (remoteSignOutContext.Skipped)
+            {
+                Logger.RemoteSignOutSkipped();
+                return false;
+            }
+
+            if (message == null)
+            {
+                return false;
+            }
+
+            // Try to extract the session identifier from the authentication ticket persisted by the sign-in handler.
+            // If the identifier cannot be found, bypass the session identifier checks: this may indicate that the
+            // authentication cookie was already cleared, that the session identifier was lost because of a lossy
+            // external/application cookie conversion or that the identity provider doesn't support sessions.
+            var sid = (await Context.Authentication.AuthenticateAsync(Options.SignInScheme))?.FindFirst("sid")?.Value
```

Comment:
Because they point to different cookies in practice?
There are far more chances to find the `sid` in the external cookie than in the application cookie, specially since Identity doesn't preserve the claims between the two cookies.

Or maybe I don't get the point of having two separate schemes :open_mouth: